File: array/remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {

        // init
        int len = nums.size();
        int pointer = 1; 
        int temp = nums[0];

        // move unique elements to the front
        for (int i = 1; i < len; i ++){
            if (nums[i] != temp) {
                nums[pointer] = nums[i];
                pointer ++;
                temp = nums[i];
            }
        }

        // return the number of unique elements
        return pointer;
    }
};

File: array/remove-duplicates-from-sorted-array.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """

        # init
        pointer = 0
        temp = None

        # move unique elements to the front
        for i in range(len(nums)):
            if nums[i] != temp:
                temp = nums[i]
                nums[pointer] = nums[i]
                pointer += 1

        # return the number of unique elements
        return pointer